app/app.go

```diff
@@ -246,18 +246,20 @@ func NewApp(
 		// akash
 		deployment.NewAppModule(
 			app.keeper.deployment,
+			app.keeper.acct,
 			app.keeper.market,
 			app.keeper.bank,
 		),
 
 		market.NewAppModule(
 			app.keeper.market,
+			app.keeper.acct,
 			app.keeper.deployment,
 			app.keeper.provider,
 			app.keeper.bank,
 		),
 
-		provider.NewAppModule(app.keeper.provider, app.keeper.bank),
+		provider.NewAppModule(app.keeper.provider, app.keeper.acct, app.keeper.bank),
```

Comment:
we should not expose/add new dependencies just for simulation sake. 

Provider/Deployment/Market doesn't need account keeper access. So, we should find a way to eliminate it.